graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/EntityModel.java

```diff
@@ -155,42 +155,33 @@ public boolean isFederated() {
 
   /**
    * Validates a set of conditions to ensure that they will produce a valid CQL query that doesn't
-   * require ALLOW FILTERING.
+   * require ALLOW FILTERING for this entity.
    *
    * @return an optional error message if the validation failed, empty otherwise.
    */
-  public Optional<String> validateConditions(Collection<WhereConditionModel> conditions) {
+  public Optional<String> validateNoFiltering(Collection<WhereConditionModel> conditions) {
```

Comment:
Renamed for clarity. In `WhereConditionModelBuilder` we validate that argument types match their fields, and that the predicates are allowed. Here we validate that the set of conditions forms a valid query.